enterprise/server/remote_execution/executor/executor.go

```diff
@@ -92,16 +90,13 @@ func timevalDuration(tv syscall.Timeval) time.Duration {
 	return time.Duration(tv.Sec)*time.Second + time.Duration(tv.Usec)*time.Microsecond
 }
 
-func parseTimeout(timeout *durationpb.Duration, maxDuration time.Duration) (time.Duration, error) {
+func parseTimeout(timeout *durationpb.Duration) (time.Duration, error) {
 	if timeout == nil {
-		if maxDuration == 0 {
-			return infiniteDuration, nil
-		}
```

Comment:
I think this check was not needed because it could only be true if `time.Until(currentDeadline)` on line 248 happened to return exactly 0, and even in that case, the context will already be done so it doesn't matter what timeout we return from this func.